internal/gcsx/reader.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcsx
+
+import (
+	"context"
+	"errors"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/gcs"
+)
+
+// FallbackToAnotherReader is returned when data could not be retrieved
+// from the current reader, indicating that the caller should attempt to fall back
+// to an alternative reader.
+var FallbackToAnotherReader = errors.New("fallback to another reader is required")
+
+// GCSReaderReq represents the request parameters needed to read a data from a GCS object.
+type GCSReaderReq struct {
```

Comment:
Please dont use the short names (Req). Either use the full name or omit the work entirely. rename this to GCSReaderRequest.